src/frame/endpoints/GroupApiEndpoints.ts

```diff
@@ -28,6 +28,15 @@ export interface GroupKeyUpdateResponse {
     needsRotation: boolean;
 }
 
+export interface GroupPublicKeyCache {
+    [groupId: string]: GroupPublicKeyObject;
+}
+
+export interface GroupPublicKeyObject {
+    id: string;
```

Comment:
Just matching the response object originally as much as possible.

Reply:
Seems like you could create a response object with this shape without having to store the id twice in the hashmap.

Reply:
I can. I thought the improvement in space efficiency wasn't worth the extra type and having it be different. I can do it though if you feel strongly about this.

Reply:
I would definitely prefer it, although I am probably worrying about something that isn't such a big deal. Won't need an extra type - the contents of the map will just be `PublicKey<string>`, right?

Reply:
Well, the problem is that each time you want to return it, you have to make the new object anyway. It basically complicates all the calls _to_ the cache instead of just stashing the right shape in the cache. (at least how I understand it). @skeet70 can clarify further if I'm wrong.

Reply:
@BobWall23 no, I'll still abstract with a type just to make future changes more straightforward instead of duplicating `PublicKey<string>` in a few places.

@coltfred it's not all that more complicated at the call site (`cacheHits.push(cache[id])` vs `cacheHits.push({id: id, ...cache[id]}`) but it does mean the extra type and changing the current one to an extension of that type. That does mean making the object each time, instead of returning a reference to the one in the cache, but that's probably safer given how people use js? 

I don't really think this matters much either way, so I just made the change. I think it was more straightforward/maintainable before, but we'll see what future me thinks.